### PR TITLE
2451 DynInvokeSchedule split & rename

### LINKED ISSUE #2451: Split and rename DynInvokeSchedule from `dynamo0p3.py`
sub issue of #2235

### PR DESCRIPTION
**Pull request by oakleybrunt**
closes #2451

### CODE REVIEW

**Comment by oakleybrunt** [comment]
@TeranIvy - tests now passing, ready for review :)

**Review by arporter — CHANGES_REQUESTED** [review]
Good job. I only managed to find one outstanding mention of DynInvokeSchedule:

    ./src/psyclone/tests/psyir/tools/call_tree_utils_test.py:    # This will return three schedule - the DynInvokeSchedule, and two

Ref. guide still builds fine, as does the UG. I'll fire off the integration tests but as these don't currently work for LFRic, I'll need independent verification from you that you tested this branch with LFRic :-)

File: src/psyclone/domain/lfric/lfric_invoke_schedule.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2017-2023, Science and Technology Facilities Council.
```
[the remark this answers is not in the document]
Comment (oakleybrunt) [reply src/psyclone/domain/lfric/lfric_invoke_schedule.py:4]: Wow hahaha

**Review by arporter — COMMENTED** [review]
All requested changes have been made (thanks) and integration tests were fine.
@oakleybrunt please could you confirm that you've tested LFRic with this branch?
Note: reference guide still builds fine.

**Comment by oakleybrunt** [comment]
> All requested changes have been made (thanks) and integration tests were fine. @oakleybrunt please could you confirm that you've tested LFRic with this branch? Note: reference guide still builds fine.

@arporter I have not! And embarrasingly I can't remember how to test specific branches of PSyclone with LFRic... can you help?

**Comment by arporter** [comment]
@oakleybrunt, re. building with LFRic, Sergi is hopeful that he will have the integration tests fixed today so we can just wait for them to come back :-)

**Comment by arporter** [comment]
I've just triggered the integration tests. If they are OK then I'll proceed to merge.

**Comment by oakleybrunt** [comment]
> I've just triggered the integration tests. If they are OK then I'll proceed to merge.

Thanks Andy, sorry for not getting to the LFRic testing myself.

**Comment by arporter** [comment]
Everything was green. will proceed to merge.

**Review by arporter — APPROVED** [review]
All good now.
Just waiting on final CI.

**Pull request merged**